### DISCUSSION TITLE
Using Tcp transport, readers nerver be notified that writers  are lost or disconnected,after discovery server and writer process were killed.

### DISCUSSION THREAD

**Discussion by chunyisong** [eProsima/Fast-DDS#4034] [Q&A] [answered]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Using Tcp transport, when discovery server and writer process killed,after keep_alive_frequency_ms,the readers should be notified writers are lost or disconnected and drop matching status.

### Current behavior

Using Tcp transport, when discovery server and writer process killed,the readers nerver be notified writers  are lost or disconnected ,and still they are matching after keep_alive_frequency_ms.

### Steps to reproduce

1. Using Tcp transport, start a discovery server
2. Using Tcp transport, start some writers publishing Hello topic,and connect to discovery server.
3. Using Tcp transport, start some readers subscribing Hello topic,and connect to discovery server.
4. Waiting for some time ,when all readers and writers matched,first kill discovery server process,and then kill all writers processes
5. Waiting for  keep_alive_frequency_ms,reader are still  get matched with the killed writers,and No message to notify connection are disconnected.

### Fast DDS version/commit

FastDDS [v2.11.2]

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

TCPv4

**Comment by EduPonz** [comment, marked as answer]
Hi @chunyisong,

I'm afraid what you are experiencing is a design decision regarding Discovery Server. When using DS, the participant liveliness is kept by the Server participants, so the Clients do not keep it between them (this is done to reduce discovery traffic). That being said, you can configure the [LivelinessQosPolicy](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/standardQosPoli-cies.html#livelinessqospolicy) of your DataWriters to `MANUAL_BY_TOPIC`, as the writer liveliness protocol in that case is kept between writers and readers. 

Alternatively, a year ago we did a proof of concept for having a mode in which the participant lease is kept between DS clients (you can find it [here](feature/p2p_discovery_implementation), although I cannot guarantee that it's functional); you could contact [eProsima's commercial support]([support@eprosima.com](mailto:support%40eprosima.com)) to move that feature forward.

In any case, I'm moving this issue to the appropriate discussion forum according to the policies outlined our [contribution guidelines](https://github.com/eProsima/Fast-DDS/blob/feature/p2p_discovery_implementation/CONTRIBUTING.md).
